python/astro_metadata_translator/translator.py

```diff
@@ -977,7 +977,8 @@ def read_all_headers(cls, filename, primary=None):
             # Prevent circular import by deferring
             from .file_helpers import read_basic_metadata_from_file
 
-            yield read_basic_metadata_from_file(filename, 0)
+            # Merge primary and secondary header if they exist.
+            yield read_basic_metadata_from_file(filename, -1)
```

Comment:
This contradicts the documented behavior, which suggests that if there are multiple headers you get multiple results. Does the description need to be updated to mention merging?

Reply:
I have updated the documentation.